### DISCUSSION TITLE
Do not know how to serialize a BigInt Next14

### DISCUSSION THREAD

**Discussion by Marcos-Iorio** [prisma/prisma#23189] [Q&A]
### Bug description

I cannot use JSON.Stringify in next14 and Prisma 5.8.0

This is how i'm serializing my code

```ts
const updateRecipe = async (order: Order | undefined): Promise<any> => {
  //Serializa el BigInt a String ya que stringify no puede manejar bigints
  (BigInt.prototype as any).toJSON = function () {
    return this.toString();
  };

  try {
    const response = await fetch(`/api/orders/update`, {
      method: "POST",
      body: JSON.stringify(order),
    });

    if (response.status === 200) {
      return await response.json();
    } else {
      throw new Error();
    }
  } catch (error) {
    return error;
  }
};
```

And this is the route that handles the db call:

```ts
import { prisma } from "@/lib/prisma";

async function POST(req: Request) {
  const body = await req.json();

  const formattedDrugs = body.drugs.map((drug: string) => BigInt(drug));

  try {
    const updatedRecipe = await prisma.orders.update({
      where: {
        id: body.id,
      },
      data: {
        givenAt: new Date(
          body.givenAt.replace(/(\d{2})-(\d{2})-(\d{4})/, "$3-$2-$1")
        ),
        retiredAt:
          body.retiredAt === null || body.retiredAt === ""
            ? null
            : new Date(
                body.retiredAt.replace(/(\d{2})-(\d{2})-(\d{4})/, "$3-$2-$1")
              ),
        boughtAt:
          body.boughtAt === null || body.boughtAt === ""
            ? null
            : new Date(
                body.boughtAt.replace(/(\d{2})-(\d{2})-(\d{4})/, "$3-$2-$1")
              ),
        patientId: BigInt(body.patientId),
        drugs: formattedDrugs,
      },
    });

    if (updatedRecipe) {
      return Response.json({
        message: "Receta actualizado exitosamente!",
        status: 200,
        data: updatedRecipe,
      });
    }
  } catch (e) {
    throw e;
  }
}

export { POST };
```

I already read all the posts of this but those doesn't solve my problem, i don't know what to do!

### Expected behavior

It should update the record in the db all the times.

Now after a few tries of working it's start failing

### Prisma information
```prisma
model Orders {
  id BigInt @id @default(autoincrement())
  givenAt DateTime @db.Date
  retiredAt DateTime? @db.Date
  boughtAt DateTime? @db.Date
  patient Patients @relation(fields: [patientId], references: [id])
  patientId BigInt
  drugs   BigInt[] @default([])
}
```

### Environment & setup
- OS: Windows 10
- Database: PostgreSQL in Vercel
- Node.js version: v18.18.2

### Prisma Version
```
5.8.0
Studio: 0.497.0
```

**Comment by nurul3101** [comment]
Hey @Marcos-Iorio 👋 

Can you try the solution suggested in this StackOverflow Answer for the same problem and check if that fixes the issue for you?
- https://stackoverflow.com/a/76013128/4154062

**Comment by tetthys** [comment]
https://stackoverflow.com/a/77529441/20164678

It is a simple issue.

Try this.

It works 100%

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
